compiler/rustc_lint/src/lib.rs

```diff
@@ -36,6 +36,8 @@
 #![feature(let_else)]
 #![feature(never_type)]
 #![recursion_limit = "256"]
+// #![deny(rustc::diagnostic_outside_of_impl)]
+// #![deny(rustc::untranslatable_diagnostic)]
```

Comment:
What is preventing these from being uncommented? I'm happy to merge what you have if these can be uncommented or removed (if there's still part of the crate to do, for example).

Reply:
Okay, lints can be migrated just like regular errors using `#[derive(LintDiagnostic)]` and [`emit_spanned_lint`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.emit_spanned_lint).